tuf/client/updater.py

```diff
@@ -1651,32 +1737,67 @@ def _update_fileinfo(self, metadata_filename):
       None.
 
     <Side Effects>
-      The file details of 'metadata_filename' is calculated and
-      stored in 'self.fileinfo'.
+      The version number of 'metadata_filename' is calculated and stored in its
+      corresponding entry in 'self.versioninfo'.
 
     <Returns>
       None.
     """
         
     # In case we delayed loading the metadata and didn't do it in
-    # __init__ (such as with delegated metadata), then get the file
+    # __init__ (such as with delegated metadata), then get the version 
     # info now.
        
     # Save the path to the current metadata file for 'metadata_filename'.
     current_filepath = os.path.join(self.metadata_directory['current'],
                                     metadata_filename)
-    # If the path is invalid, simply return and leave fileinfo unset.
+    # If the path is invalid, simply return and leave versioninfo unset.
     if not os.path.exists(current_filepath):
-      self.fileinfo[metadata_filename] = None
+      self.versioninfo[metadata_filename] = None
       return
    
-    # Extract the file information from the actual file and save it
-    # to the fileinfo store.
-    file_length, hashes = tuf.util.get_file_details(current_filepath)
-    metadata_fileinfo = tuf.formats.make_fileinfo(file_length, hashes)
-    self.fileinfo[metadata_filename] = metadata_fileinfo
-  
- 
+    # Extract the version information from the trusted snapshot role and save
+    # it to the 'self.versioninfo' store.
+    if metadata_filename == 'timestamp.json':
+      trusted_versioninfo = \
+        self.metadata['current']['timestamp']['version']
+
+    # When updating snapshot.json, the client either (1) has a copy of
+    # snapshot.json, or (2) in the process of obtaining it by first downloading
```

Comment:
Resolved [here in 321](https://github.com/theupdateframework/tuf/pull/321/files#diff-852078e79350a80482b37e47cae64ed3R1769).